client/graphite/read.go

```diff
@@ -192,12 +193,27 @@ func (c *Client) targetToTimeseries(ctx context.Context, target string, from str
 				"path", renderResponse.Target, "prefix", c.cfg.DefaultPrefix, "err", err)
 			return nil, err
 		}
-		for _, datapoint := range renderResponse.Datapoints {
-			timstampMs := datapoint.Timestamp * 1000
+		for j, datapoint := range renderResponse.Datapoints {
+			timestampMs := datapoint.Timestamp * 1000
 			if datapoint.Value == nil {
 				continue
 			}
-			ts.Samples = append(ts.Samples, &prompb.Sample{Value: *datapoint.Value, Timestamp: timstampMs})
+			ts.Samples = append(ts.Samples, &prompb.Sample{Value: *datapoint.Value, Timestamp: timestampMs})
+
+			// If not last point and interpolation enable, linear interpolate intermediate points.
+			if (j+1) < len(renderResponse.Datapoints) && c.cfg.Read.MaxPointDelta != time.Duration(0) {
+				intervalSecond := int64(c.cfg.Read.MaxPointDelta.Seconds())
+				nextDatapoint := renderResponse.Datapoints[j+1]
+				if nextDatapoint.Value == nil {
+					continue
+				}
+
+				variation := (*nextDatapoint.Value - *datapoint.Value) / float64(nextDatapoint.Timestamp-datapoint.Timestamp)
+				for timestamp := datapoint.Timestamp + intervalSecond; timestamp < nextDatapoint.Timestamp; timestamp += intervalSecond {
```

Comment:
Maybe something like that would be more clear ?

```
for i := 0; i < delta / interval; i++ {
timestamp := i * interval
value := *datapoint.Value + float64(timestamp-datapoint.Timestamp)*variation
}
```

Reply:
Done.